registry/reference.go

```diff
@@ -56,33 +56,65 @@ type Reference struct {
 	Reference string
 }
 
-// ParseReference parses a string into a artifact reference.
-// If the reference contains both the tag and the digest, the tag will be
-// dropped.
-// Digest is recognized only if the corresponding algorithm is available.
-func ParseReference(raw string) (Reference, error) {
-	parts := strings.SplitN(raw, "/", 2)
+// ParseReference parses a string (artifact) into an `artifact reference`.
```

Comment:
<img width="719" alt="image" src="https://user-images.githubusercontent.com/32161882/186350262-1059eeb1-1dde-4084-9007-35ef0d06975e.png">

The doc looks nice.